static/components/PoiCard.js

```diff
@@ -0,0 +1,98 @@
+import React from 'preact-compat'
+// Component
+import Rank from './Rank'
+import Power from './Power'
+import Comments from './Comments'
+
+class PoiCard extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      from: {},
+      to: {},
+      unit: ['meters', 'km']
+    }
+    // cf: https://facebook.github.io/react/docs/handling-events.html
+    this.hoverCard = this.hoverCard.bind(this)
+    this.clickCard = this.clickCard.bind(this)
+  }
+
+  componentDidMount() {
+    this.setState({
+      from: [this.props.details.position.latitude, this.props.details.position.longitude],
+      to: [this.props.userLat, this.props.userLng]
+    })
+    if(this.props.details._id === getHash()) {
+      highlight(this.props.details._id)
+      superCard(this.props.details._id)
+      firstCard(this.props.details._id)
+    }
+  }
+
+  calculateDistance(from, to, unit) {
+    if(to[0] && to[1]) {
+      const R = 6371e3 // metres
+      const lat1 = this.state.from[0] * Math.PI / 180
+      const lat2 = this.state.to[0] * Math.PI / 180
+      const gapLat = ((lat2 * Math.PI / 180 )-(lat1 * Math.PI / 180))
+      const gapLng = ((this.state.to[1] * Math.PI / 180)-(this.state.from[1] * Math.PI / 180))
+      const a = Math.sin(gapLat/2) * Math.sin(gapLat/2) + Math.cos(lat1) * Math.cos(lat2) * Math.sin(gapLng/2) * Math.sin(gapLng/2)
+      const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a))
+
+      let distance = Math.trunc(R * c)
+      
+      if(unit) {
+        return this.renderDistance(distance, unit)
```

Comment:
This should be outside of this function.
Just consider that `calculateDistance()` would return a number and that's it. It just process a mathematical calculation.
Aside, out of the function, totally apart of it, you would have `renderDistance(distance, unit)` that would render (= preparing for display) a distance with its unit.
So you'd use it like that:

    distance = calculateDistance(…)
    renderDistance(distance)